gector/gec_predictor.py

```diff
@@ -143,15 +153,77 @@ def predict_batch_instance(self, instances: List[Instance]) -> List[JsonDict]:
                 - corrected_words
             For an explanation of each of these see `Seq2Labels.decode()`.
         """
-        for i in range(self._iterations):
-            outputs = self._model.forward_on_instances(instances)
-            corrected_instances = []
-            for output in outputs:
-                tokens = [Token(word) for word in output["corrected_words"]]
-                instance = self._dataset_reader.text_to_instance(tokens)
-                corrected_instances.append(instance)
-            instances = corrected_instances
-        return sanitize(outputs)
+
+        # Make deep copy of batch
+        final_batch = instances[:]
+
+        final_outputs = [None] * len(instances)
+
+        prev_preds_dict = {
+            id: [final_batch[id].fields["tokens"].tokens]
+            for id in range(len(final_batch))
+        }
+
+        short_ids = [
+            id
+            for id in range(len(final_batch))
+            if len(final_batch[id].fields["tokens"].tokens) < 4
+        ]
```

Comment:
It seems a little wasteful to iterate over `final_batch` twice? Couldn't we replace these two statements with a simple for loop and do both things in the same loop?